### ISSUE TITLE
"using" should emit a warning when Dispose() is marked as obsolete

### ISSUE THREAD

**Issue by jhinder** [dotnet/roslyn#31710]
**Version Used**: VS 16.0 Preview 1

**Steps to Reproduce**:

1. Copy the code below into a C# file.

```c#
class C : IDisposable
{
    public void M()
    {
        using (var c = new C())
        {
        }
    }

    [Obsolete("Type won't be disposable in v2.0")]
    public void Dispose() { }
}
```

**Expected Behavior**: A warning CS0612/CS0618 should be reported, similar to when you'd call the `Dispose` method directly. One possible location for the warning could be the using statement's variable declaration/expression.

**Actual Behavior**: No warning is reported.

**Comment by gafter** [comment]
The compiler does not bind anything to `C.Dispose`. It binds to `IDisposable.Dispose`, which is not obsolete.
Therefore no obsolete diagnostic is expected. See https://github.com/dotnet/csharplang/blob/master/spec/statements.md#the-using-statement for details.

**Comment by alrz** [comment]
So with pattern-based using, if we *remove* `IDisposable` we should get a warning? That would be weird.
